src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -778,6 +778,11 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
             if (hasNaturalType)
             {
                 tupleTypeOpt = TupleTypeSymbol.Create(node.Location, elements, locations, elementNamesArray, this.Compilation, node, diagnostics);
+
+                if (ShouldCheckConstraints)
+                {
+                    tupleTypeOpt.TupleUnderlyingType.CheckConstraints(this.Compilation, this.Conversions, node.Location, diagnostics);
```

Comment:
>CheckConstraints [](start = 53, length = 16)

I think it would be better if CheckConstraints handled tuple type symbols itself.